flake.nix

```diff
@@ -0,0 +1,63 @@
+{
+  description = "nixos-hardware";
+
+  outputs = { self }: {
+    nixosModules = {
+      dell-aspire-4810t = import ./acer/aspire/4810t;
+      apple-macbook-air-3 = import ./apple/macbook-air/3;
+      apple-macbook-air-4 = import apple/macbook-air/4;
+      apple-macbook-air-6 = import apple/macbook-air/6;
+      apple-macbook-air-10-1 = import ./apple/macbook-pro/10-1;
+      apple-macbook-air-12-1 = import ./apple/macbook-pro/12-1;
+      beagleboard-pocketbeagle = import ./beagleboard/pocketbeagle;
+      dell-latitude-3480 = import ./dell/latitude/3480;
+      dell-e7240 = import ./dell/e7240;
+      dell-xps-13-7390 = import ./dell/xps/13-7390;
+      dell-xps-13-9360 = import ./dell/xps/13-9360;
+      dell-xps-13-9370 = import ./dell/xps/13-9370;
+      dell-xps-13-9380 = import ./dell/xps/13-9380;
+      dell-xps-15-7590 = import ./dell/xps/15-7590;
+      dell-xps-15-9550 = import ./dell/xps/15-9550;
+      dell-xps-15-9560 = import ./dell/xps/15-9560;
+      dell-xps-15-9560-intel = import ./dell/xps/15-9560/intel;
+      dell-xps-15-9560-nvidia = import ./dell/xps/15-9560/nvidia;
+      google-pixelbook = import ./google/pixelbook;
+      inversepath-usbarmory = import ./inversepath/usbarmory;
+      lenovo-ideapad-z510 = import ./lenovo/ideapad/z510;
+      lenovo-thinkpad-e495 = import ./lenovo/thinkpad/e495;
+      lenovo-thinkpad-l13 = import ./lenovo/thinkpad/l13;
+      lenovo-thinkpad-p53 = import ./lenovo/thinkpad/p53;
+      lenovo-thinkpad-t410 = import ./lenovo/thinkpad/t410;
+      lenovo-thinkpad-t420 = import ./lenovo/thinkpad/t420;
+      lenovo-thinkpad-t430 = import ./lenovo/thinkpad/t430;
+      lenovo-thinkpad-t440s = import ./lenovo/thinkpad/t440s;
+      lenovo-thinkpad-t440p = import ./lenovo/thinkpad/t440p;
+      lenovo-thinkpad-t450s = import ./lenovo/thinkpad/t450s;
+      lenovo-thinkpad-t460s = import ./lenovo/thinkpad/t460s;
+      lenovo-thinkpad-t470s = import ./lenovo/thinkpad/t470s;
+      lenovo-thinkpad-t480s = import ./lenovo/thinkpad/t480s;
+      lenovo-thinkpad-t490 = import ./lenovo/thinkpad/t490;
+      lenovo-thinkpad-t495 = import ./lenovo/thinkpad/t495;
+      lenovo-thinkpad-x140e = import ./lenovo/thinkpad/x140e;
+      lenovo-thinkpad-x220 = import ./lenovo/thinkpad/x220;
+      lenovo-thinkpad-x230 = import ./lenovo/thinkpad/x230;
+      lenovo-thinkpad-x250 = import ./lenovo/thinkpad/x250;
+      lenovo-thinkpad-x260 = import ./lenovo/thinkpad/x260;
+      lenovo-thinkpad-x270 = import ./lenovo/thinkpad/x270;
+      lenovo-thinkpad-x280 = import ./lenovo/thinkpad/x280;
```

Comment:
I wonder what would be preferable, `lenovo-thinkpad-x280` or `lenovo.thinkpad.x280`.

Reply:
I am not sure if we can use `lenovo.thinkpad.x280` i.e. if flakes support nesting under `nixosModules`. The current implementation seems like the safest choice to me.

Reply:
Nesting is not supported according to: https://github.com/NixOS/rfcs/pull/49#discussion_r450275707